tests/dragonfly/replication_test.py

```diff
@@ -1696,3 +1696,44 @@ async def test_network_disconnect_small_buffer(df_local_factory, df_seeder_facto
     master.stop()
     replica.stop()
     assert master.is_in_logs("Partial sync requested from stale LSN")
+
+
+# @pytest.mark.slow
+@pytest.mark.asyncio
+async def test_client_pause_with_replica(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(proactor_threads=4)
+    replica = df_local_factory.create(proactor_threads=4)
+    df_local_factory.start_all([master, replica])
+
+    seeder = df_seeder_factory.create(port=master.port)
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    fill_task = asyncio.create_task(seeder.run(target_ops=3000))
+
+    # Give the seeder a bit of time.
+    await asyncio.sleep(1)
+    # block the seeder for 2 seconds
```

Comment:
Let's avoid specific numbers for timing related tests. You can just start the seeder without a target and then call stop() and await the task as you already do at the end